csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -35,6 +36,8 @@ public abstract class HiveServer2Connection : AdbcConnection
         internal TTransport? transport;
         internal TCLIService.Client? client;
         internal TSessionHandle? sessionHandle;
+        private string? _vendorVersion;
```

Comment:
Should these lazy-instantiated fields be atomics?

Reply:
Now setting the values using `Lazy<>` in a thread-safe way. Unlikely to be too expensive on a once-per-connection call.